test/src/server.py

```diff
@@ -0,0 +1,1558 @@
+        Execute command on the server.
+
+        If the server is not localhost the command is executed over SSH.
+
+        Parameters
+        ----------
+        command : str
+            The command to execute.
+
+        Returns
+        -------
+        str
+            The output of the command.
+
+        See Also
+        --------
+        __exec_local : Execute a command on the localhost.
+        __exec_ssh : Execute a command on the server over SSH.
+
+        Example
+        -------
+        >>> print(server.exec('ls -l'))
+        .bashrc
+        """
+        debug(f'Executing command on {self.log_name()}: {command}')
+        if self.localhost:
+            return self.__exec_local(command)
+        else:
+            return self.__exec_ssh(command)
+
+    def whoami(self: 'Server') -> str:
+        """
+        Get the user name.
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        str
+            The user name.
+        """
+        return self.exec('whoami').strip()
+
+    def gethome(self: 'Server') -> str:
+        """
+        Get the home directory.
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        str
+            The home directory.
+        """
+        return self.exec('echo $HOME').strip()
+
+    def isfile(self: 'Server', path: str) -> bool:
+        """
+        Check if a file exists.
+
+        Parameters
+        ----------
+        path : str
+            The path to the file.
+
+        Returns
+        -------
+        bool
+            True if the file exists.
+        """
+        return self.exec(f'test -f {path} && echo true || echo false'
+                         ).strip() == 'true'
+
+    def tmux_new(self: 'Server', session_name: str, command: str) -> None:
+        """
+        Start a tmux session on the server.
+
+        Parameters
+        ----------
+        session_name : str
+            The name of the session.
+        command : str
+            The command to execute.
+
+        Returns
+        -------
+
+        See Also
+        --------
+        exec : Execute command on the server.
+        tmux_kill : Stop a tmux session on the server.
+        tmux_send_keys : Send keys to a tmux session on the server.
+        """
+        _ = self.exec(f'tmux new-session -s {session_name} -d "{command}"')
+
+    def tmux_kill(self: 'Server', session_name: str) -> None:
+        """
+        Stop a tmux session on the server.
+
+        Parameters
+        ----------
+        session_name : str
+            The name of the session.
+
+        Returns
+        -------
+
+        See Also
+        --------
+        exec : Execute command on the server.
+        tmux_new : Start a tmux session on the server.
+        tmux_send_keys : Send keys to a tmux session on the server.
+        """
+        _ = self.exec('tmux list-sessions | cut -d ":" -f 1 ' +
+                      f'| grep {session_name} | xargs tmux kill-session -t')
```

Comment:
how about you dont kill my tmux sessions. Add use a static socket like `tmux -L vmux.sock` for all tmux interactions to seperate autotests tmux interactions from the user ones. 
Example: https://github.com/Mic92/vmsh/blob/cfbb612d5f5615b194fbacd9b4a32e9816eac3b4/tests/qemu.py#L303

Reply:
Resolved in 7e8ad45900f6a58bd6f5a6dbb77df5ae580ca398 ... see commit message for details, tldr: set `tmux_socket` parameters in your `autotest.cfg`.